pkg/ui/workspaces/cluster-ui/src/databaseTablePage/util.tsx

```diff
@@ -0,0 +1,226 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+import {
+  DATE_FORMAT,
+  DATE_FORMAT_24_TZ,
+  EncodeDatabaseTableUri,
+  EncodeDatabaseUri,
+  EncodeUriName,
+  minDate,
+  performanceTuningRecipes,
+} from "../util";
+import { Timestamp } from "../timestamp";
+import React from "react";
+import { Moment } from "moment-timezone";
+import { DatabaseTablePageDataDetails, IndexStat } from "./databaseTablePage";
+import { CircleFilled } from "../icon";
+import { Tooltip } from "antd";
+import "antd/lib/tooltip/style";
+import { Anchor } from "../anchor";
+import classNames from "classnames/bind";
+import styles from "./databaseTablePage.module.scss";
+import { QuoteIdentifier } from "../api/safesql";
+import IdxRecAction from "../insights/indexActionBtn";
+import * as format from "../util/format";
+import { Link } from "react-router-dom";
+import { Search as IndexIcon } from "@cockroachlabs/icons";
+import { Breadcrumbs } from "../breadcrumbs";
+import { CaretRight } from "../icon/caretRight";
+const cx = classNames.bind(styles);
+
+export const NameCell = ({
+  indexStat,
+  showIndexRecommendations,
+  tableName,
+  isCockroachCloud,
+}: {
+  indexStat: IndexStat;
+  showIndexRecommendations: boolean;
+  tableName: string;
+  isCockroachCloud: boolean;
+}): JSX.Element => {
+  if (showIndexRecommendations) {
+    const linkURL = isCockroachCloud
+      ? `${location.pathname}/${indexStat.indexName}`
+      : `${tableName}/index/${EncodeUriName(indexStat.indexName)}`;
+    return (
+      <Link to={linkURL} className={cx("icon__container")}>
+        <IndexIcon className={cx("icon--s", "icon--primary")} />
+        {indexStat.indexName}
+      </Link>
+    );
+  }
+  return (
+    <>
+      <IndexIcon className={cx("icon--s", "icon--primary")} />
+      {indexStat.indexName}
+    </>
+  );
+};
+
+export const DbTablesBreadcrumbs = ({
+  tableName,
+  schemaName,
+  databaseName,
+  isCockroachCloud,
+}: {
+  tableName: string;
+  schemaName: string;
+  databaseName: string;
+  isCockroachCloud: boolean;
```

Comment:
Prefer using the context in the function body instead of prop drilling.

Reply:
Yup, forgot to add that change to this PR.